doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,233 @@
+        public virtual Client2 GetClient2Client(string endpoint)
+        {
+            Argument.AssertNotNull(endpoint, nameof(endpoint));
+
+            return new Client2(ClientDiagnostics, _pipeline, _keyCredential, endpoint);
+        }
+    }
+}
+
+// SubClient: Client1 Generated\Client1.cs
+namespace Azure.Service.SubClients
+{
+-   public partial class Client1Client
++   public partial class Client1
+    {
+-       protected Client1Client(){}
++       protected Client1(){}
+-       public Client1Client(string endpoint, string cachedParameter, AzureKeyCredential credential, SubClientsClientOptions options = null){}
++       internal Client1(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, AzureKeyCredential keyCredential, string endpoint, string cachedParameter){}
+    }
+}
+
+//subclient: Client2 Generated\Client2.cs
+namespace Azure.Service.SubClients
+{
+-   public partial class Client2Client
++   public partial class Client2
+    {
+-       protected Client2Client()
++       protected Client2()
+-       public Client2Client(string endpoint, AzureKeyCredential credential, SubClientsClientOptions options = null){}
++       internal Client2(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, AzureKeyCredential keyCredential, string endpoint) {}
+    }
+}
+
+```
+
+</details>
+
+### Client hierarchy
+
+Each client can be made a subclient of another client (without circular dependency) using CodeGenClientAttribute.ParentClient parameter.
+Since CodeGenClientAttribute is applied to the client type, user will explicitly specify new client type name. According to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients), type should have a name without "Client" suffix and an internal constructor. A factory method will be added to the type, specified in ParentClient.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\Client1Client.cs
+namespace Azure.Service.SubClients
+{
+    public partial class Client1Client
+    {
+        public Client1Client(string endpoint, string cachedParameter, AzureKeyCredential credential, SubClientsClientOptions options = null){}
+    }
+}
+
+//Generated\Client2Client.cs
+namespace Azure.Service.SubClients
+{
+    public partial class Client2Client
+    {
+        public Client2Client(string endpoint, AzureKeyCredential credential, SubClientsClientOptions options = null){}
+    }
+}
+```
+
+**Add client hierarchy customization (Customizations.cs):**
+
+```C#
+
+//Customizations.cs
+using Azure.Core;
+
+namespace Azure.Service.SubClients
+{
+    [CodeGenClient("Client2Client", ParentClient = typeof(Client1Client))]
+    public partial class Client2 { }
+}
+```
+
+**Generated code after:**
+
+```diff
+namespace Azure.Service.SubClients
+{
+-   public partial class SubClientsClientOptions : ClientOptions{}
++   public partial class Client1ClientOptions : ClientOptions{}
```

Comment:
Is there any configuration change that would cause this diff in the generated code?